src/openrct2/ride/ride.c

```diff
@@ -8681,6 +8681,14 @@ void fix_invalid_vehicle_sprite_sizes()
             while (rideSpriteIndex != SPRITE_INDEX_NULL) {
                 rct_vehicle *vehicle = GET_VEHICLE(rideSpriteIndex);
                 rct_ride_entry_vehicle *vehicleEntry = vehicle_get_vehicle_entry(vehicle);
+
+                if (vehicle == NULL ||
+                    vehicleEntry == NULL ||
+                    vehicleEntry == (rct_ride_entry_vehicle*)-1)
+                {
+                    continue;
```

Comment:
Shouldn't this `break` instead? `rideSpriteIndex` is not getting updated in the `while` loop, so you would end up selecting the very same vehicle over and over again in an endless loop.